--- /lib/partman/init.d/30parted	2019-02-04 19:13:42.000000000 +0300
+++ 30parted	2019-04-01 22:55:48.022861733 +0300
@@ -196,7 +196,15 @@
 				IFS="$NL"
 				for line in $mounts; do
 					restore_ifs
-					if [ "$path" = "${line%% *}" ]; then
+					# for efi compatible bootable discs there are multiple
+					# partitions on the "/cdrom drive", so need to strip
+					# the partition number and check the device itself
+					# to see if it is mounted as "/cdrom"
+					#
+					# later, "/cdrom" will be removed from partman
+					# so not considered for installation
+					pathdevice=$(echo "$path" | sed s/[0-9]*//g)
+					if [ "$pathdevice" = "${line%% *}" ]; then
 						mp="${line#* }"
 						break
 					fi
@@ -213,21 +221,23 @@
 			done
 			close_dialog
 			if [ "$instparts" ]; then
-				if [ -z "$seen_unmounted" ] && \
-				   longint_le "$free" 16777216; then
+				# regardless of free size, drive containing "/cdrom" should
+				# be removed
+				#if [ -z "$seen_unmounted" ] && \
+				#   longint_le "$free" 16777216; then
 					# The installation medium uses more
 					# or less the whole disk.
 					open_dialog CLOSE
 					close_dialog
 					cd /
 					rm -rf "$dev"
-				else
+				#else
 					# There's an installation medium
 					# here, but it doesn't use the whole
 					# disk.
-					part_warn="$instparts"
-					>installation_medium
-				fi
+				#	part_warn="$instparts"
+				#	>installation_medium
+				#fi
 			elif [ "$parts" ]; then
 				# Something other than an installation
 				# medium is mounted.
